src/java/org/apache/cassandra/config/CassandraDuration.java

```diff
@@ -27,22 +27,23 @@
 import com.google.common.primitives.Ints;
 
 /**
- * Represents a positive time duration.
+ * Represents a positive time duration. Wrapper class for Cassandra duration configuration parameters, providing to the
+ * users the opportunity to be able to provide config with a unit of their choice in cassandra.yaml as per the available
+ * options. (CASSANDRA-15234)
  */
-public final class Duration
+public final class CassandraDuration
 {
     /**
      * The Regexp used to parse the duration provided as String.
      */
-    private static final Pattern TIME_UNITS_PATTERN = Pattern.compile(("^(\\d+)([a-zA-Z]{1,2}|µs|µS)$"));
-    private static final Pattern DOUBLE_TIME_UNITS_PATTERN = Pattern.compile(("^(\\d+\\.\\d+)([a-zA-Z]{1,2}|µs|µS)$"));
+    private static final Pattern TIME_UNITS_PATTERN = Pattern.compile(("^(\\d+)(d|D|h|H|s|S|ms|MS|mS|Ms|us|US|uS|Us|µs|µS|ns|NS|nS|Nsm|M|m)"));
+    private static final Pattern DOUBLE_TIME_UNITS_PATTERN = Pattern.compile(("^(\\d+\\.\\d+)(d|D|h|H|s|S|ms|MS|mS|Ms|us|US|uS|Us|µs|µS|ns|NS|nS|Nsm|M|m)$"));
```

Comment:
It seems I introduced a typo - it is just Ns. Which reminds to add a simple unit test for all accepted formats of the units, this would have caught it....Thanks!

Reply:
or just use the toLowerCase as suggested and consider locale.ROOT the way you did it in the suggested tests. Thanks